examples/paywall-tester/src/main/java/com/revenuecat/paywallstester/ui/screens/paywall/PaywallScreen.kt

```diff
@@ -29,8 +31,42 @@ fun PaywallScreen(
                 Text(text = "Error: ${state.errorMessage}")
             }
             is PaywallScreenState.Loaded -> {
-                PaywallView(PaywallViewOptions.Builder().setOffering(state.offering).build())
+                PaywallView(
+                    PaywallViewOptions.Builder()
+                        .setOffering(state.offering)
+                        .setListener(viewModel)
+                        .build(),
+                )
+                if (state.displayCompletedPurchaseMessage) {
+                    PurchaseAlertDialog(viewModel, "Purchase was successful")
+                } else if (state.displayErrorPurchasingMessage) {
+                    PurchaseAlertDialog(viewModel, "There was a purchasing error")
```

Comment:
I might change this to reuse the dialog for restoring too. In that case the text will be set from the viewmodel. What do you think @tonidero

Reply:
see https://github.com/RevenueCat/purchases-android/pull/1247